accounts-db/src/append_vec.rs

```diff
@@ -996,18 +996,29 @@ impl AppendVec {
                         // data doesn't fit, so don't include
                         break;
                     }
-                    result.push(next.stored_size_aligned);
+                    account_sizes.push(next.stored_size_aligned);
                 }
             }
-            AppendVecFileBacking::File(_file) => {
-                for &offset in sorted_offsets {
-                    self.get_stored_account_meta_callback(offset, |stored_meta| {
-                        result.push(stored_meta.stored_size());
-                    });
+            AppendVecFileBacking::File(file) => {
+                // self.len() is an atomic load, so only do it once
+                let valid_file_len = self.len();
+                let mut buffer = [0u8; mem::size_of::<StoredMeta>()];
+                for offset in sorted_offsets {
+                    let Some(bytes_read) =
+                        read_into_buffer(file, valid_file_len, *offset, &mut buffer).ok()
+                    else {
+                        break;
+                    };
+                    let bytes = ValidSlice(&buffer[..bytes_read]);
+                    let Some((stored_meta, _next)) = Self::get_type::<StoredMeta>(bytes, 0) else {
+                        break;
+                    };
+                    let stored_size = aligned_stored_size(stored_meta.data_len as usize);
+                    account_sizes.push(stored_size);
```

Comment:
1. In the mmap case, it skipped the last element if the data.len > append_vec.len(). While for the file case, we don't do that. I think it is fine because the file-based storage won't contain incomplete accounts. To be defensive, we might want to add an assert that end_of_data_offset < self.len().
2. We are reading 64 bytes a time. It would be interesting to benchmark and see whether we get better performance by increasing the buffer size to prefetch more data on one read and reduce the number of reads. 

To be clear, item 2 is a suggestion for future work. Don't need to address in this PR.

Reply:
Note that in `get_stored_account_meta_callback`, we are using 4K page size buffer to prefetch. 
Maybe we should benchmark the PR to make sure that there is no performance degradation.

Reply:
Ah! `get_stored_account_meta_callback` we prefetch the data but *don't* use the prefetched data to save reading for future accounts! Feel free to ignore my last comments.